xchange-upbit/src/main/java/org/knowm/xchange/upbit/UpbitAdapters.java

```diff
@@ -131,27 +138,27 @@ public static Order adaptOrderInfo(UpbitOrderResponse upbitOrderResponse) {
     } else if (upbitOrderResponse.getState().equalsIgnoreCase("done")) {
       status = Order.OrderStatus.FILLED;
     }
-    OrderType type = upbitOrderResponse.getSide().equals("ask") ? OrderType.ASK : OrderType.BID;
-    BigDecimal originalAmount = upbitOrderResponse.getVolume();
-    String[] markets = upbitOrderResponse.getMarket().split("-");
-    CurrencyPair currencyPair =
-        new CurrencyPair(new Currency(markets[1]), new Currency(markets[0]));
-    String orderId = upbitOrderResponse.getUuid();
-    BigDecimal cumulativeAmount = upbitOrderResponse.getExecutedVolume();
-    BigDecimal price = upbitOrderResponse.getAvgPrice();
-    ZonedDateTime dateTime = ZonedDateTime.parse(upbitOrderResponse.getCreatedAt());
-    LimitOrder order =
-        new LimitOrder(
-            type,
-            originalAmount,
-            currencyPair,
-            orderId,
-            Date.from(dateTime.toInstant()),
-            price,
-            price,
-            cumulativeAmount,
-            upbitOrderResponse.getPaidFee(),
-            status);
-    return order;
+    return new LimitOrder(
+        upbitOrderResponse.getSide().equals("ask") ? OrderType.ASK : OrderType.BID,
```

Comment:
put the ask on the left side for NPE protection

Reply:
fixed.